internal/epoll/poller.go

```diff
@@ -138,6 +144,10 @@ func (p *Poller) Wait(events []unix.EpollEvent) (int, error) {
 			return 0, err
 		}
 
+		if n == 0 && msec != -1 {
+			return 0, os.ErrDeadlineExceeded
```

Comment:
I think we should wrap the error here, otherwise we return a naked ErrDeadlineExceeded.

```suggestion
			return 0, fmt.Errorf("poll: %w", os.ErrDeadlineExceeded)
```